tiledb/tests/test_attribute.py

```diff
@@ -63,6 +63,26 @@ def test_attribute_fill(self, dtype, fill):
                 # record type unsupported for .df
                 assert R.df[0][""].values == np.array(fill, dtype=dtype)
 
+    @pytest.mark.parametrize(
+        "dtype, fill",
+        [
+            (np.float32, np.float32(1.5)),
+            (np.float64, np.float64(2.5)),
+            (np.int32, 42),
+            (np.int64, 123),
+            (str, "new_fill"),
+            (np.dtype(bytes), b"xyz"),
+            (np.dtype("complex64"), (1.0 + 2.0j)),
+            (np.dtype("complex128"), (3.0 + 4.0j)),
+            (np.dtype("complex64"), np.complex64([1.0 + 2.0j])),
+            (np.dtype("complex128"), np.complex128([3.0 + 4.0j])),
+        ],
+    )
+    def test_fill_value_setter(self, dtype, fill):
+        attr = tiledb.Attr("test_attr", dtype=dtype)
+        attr.fill = fill
+        assert attr.fill == fill
```

Comment:
I can only assume that the `@fill.setter` annotation somehow connects this to a C API but as far as I can tell this test will pass even if it does not do so.

If you added a negative test (what happens if you try to set a number on a string attribute? I assume the `encode` raises an exception?) it will demonstrate this crucial aspect well enough, I think